src/SixLabors.Core/Memory/ArrayPoolMemoryAllocator.cs

```diff
@@ -109,6 +117,9 @@ public override IMemoryOwner<T> Allocate<T>(int length, AllocationOptions option
         /// <inheritdoc />
         public override IManagedByteBuffer AllocateManagedByteBuffer(int length, AllocationOptions options = AllocationOptions.None)
         {
+            Guard.MustBeGreaterThanOrEqualTo(length, 0, nameof(length));
+            Guard.MustBeLessThan(length, int.MaxValue, nameof(length));
```

Comment:
Is `int.MaxValue` not allowed? Otherwise we don't need this check here and in other places?

Reply:
It was throwing OOM for me, but I realized it's not specific to `int.MaxValue`, so you're right, this check doesn't make any sense, I will remove it.